src/Player.cpp

```diff
@@ -67,3 +68,25 @@ void Player::ResetHit()
 {
 	gequacked = false;
 }
+
+void Player::ScreenBounds()
+{
+	extern int WindowWidth;
+	extern int WindowHeight;
+	
+	if (collisionShape.getPosition().x > (WindowWidth - (duckyTexture.getSize().x)/2)){
+		duckySprite.setPosition((WindowWidth - (duckyTexture.getSize().x) / 2), (collisionShape.getPosition().y));
+	}
+
+	if (collisionShape.getPosition().x < 0.f){
```

Comment:
Would need to test it because I forgot where the origin is, but here you are not adding textureSize/2, so either way the Ducky will probably move unevenly across the screen (it will not stay within bounds in exact same way on both sides)